src/olympia/editors/tests/test_models.py

```diff
@@ -674,6 +676,161 @@ def setUp(self):
+            'negative_reviews': 0,
+            'past_rejection_history': 0
+        }
+        assert weight_info == expected_result
+
+    def test_calculate_weight_admin_review(self):
+        self.addon.update(admin_review=True)
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 100
+        assert weight_info['admin_review'] == 100
+
+    def test_calculate_weight_abuse_reports(self):
+        # Extra abuse report for a different add-on, does not count.
+        AbuseReport.objects.create(addon=addon_factory())
+
+        # Extra abuse report for a different user, does not count.
+        AbuseReport.objects.create(user=user_factory())
+
+        # Extra old abuse report, does not count either.
+        old_report = AbuseReport.objects.create(addon=self.addon)
+        old_report.update(created=self.days_ago(370))
+
+        # Recent abuse reports.
+        AbuseReport.objects.create(addon=self.addon)
+        AbuseReport.objects.create(addon=self.addon)
+
+        # Recent abuse report for one of the developers of the add-on.
+        author = user_factory()
+        AddonUser.objects.create(addon=self.addon, user=author)
+        AbuseReport.objects.create(user=author)
+
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 30
+        assert weight_info['abuse_reports'] == 30
+
+        # Should be capped at 100.
+        for i in range(0, 10):
+            AbuseReport.objects.create(addon=self.addon)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 100
+        assert weight_info['abuse_reports'] == 100
+
+    def test_calculate_weight_negative_reviews(self):
+        # Positive review, does not count.
+        Review.objects.create(
+            user=user_factory(), addon=self.addon, version=self.version,
+            rating=5)
+
+        # Negative review, but too old, does not count.
+        old_review = Review.objects.create(
+            user=user_factory(), addon=self.addon, version=self.version,
+            rating=1)
+        old_review.update(created=self.days_ago(370))
+
+        # Negative review on a different add-on, does not count either.
+        extra_addon = addon_factory()
+        Review.objects.create(
+            user=user_factory(), addon=extra_addon,
+            version=extra_addon.current_version, rating=1)
+
+        # Recent negative reviews.
+        Review.objects.create(
+            user=user_factory(), addon=self.addon, version=self.version,
+            rating=1)
+        Review.objects.create(
+            user=user_factory(), addon=self.addon, version=self.version,
+            rating=2)
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 4
+        assert weight_info['negative_reviews'] == 4
```

Comment:
Yeah... I'll try it, but I wasn't a fan of having to create a hundred negative reviews... and I didn't want to resort to mocking the queryset...